training/training.module

```diff
@@ -0,0 +1,378 @@
+ * @return bool
+ */
+function training_custom_access($role_name){
+  global $user;
+  return in_array($role_name, $user->roles);
+}
+
+/**
+ * Preprocessing argument. Returns the new value of the argument
+ * @param $arg - argument in URL training/menu/default_arg/%digit
+ * @return string
+ */
+
+function digit_load($arg){
+  switch ($arg) {
+    case 1:
+      $arg = 'one';
+      break;
+    case 2:
+      $arg = 'two';
+      break;
+    case 3:
+      $arg = 'three';
+      break;
+    case 99:
+      $arg = 'Almost there';
+      break;
+    default:
+      $arg = NULL;
+  }
+  return $arg;
+}
+
+/**
+ * Returns a dynamic page title
+ * string $user_name - contents user name
+ * @return string
+ */
+
+function training_menu_title_callback(){
+  global $user;
+  isset($user->name)?$user_name=$user->name:$user_name='anonymous';
+  return 'Dynamic title: username= '.$user_name;
+}
+
+
+/**
+ * Implements hook_permission()
+ * array $permission - array of permission
+ * @return array
+ */
+ function training_permission(){
+  $permission = array(
+    'train' => array(
+      'title' => 'Access protected menu example',
+      'description' => 'Use this to control access to page Permissioned Menu Item of training module',
+    ),
+    'train_theme' => array(
+      'title' => 'Access protected content',
+      'description' => 'Use this to control access to page Access Theme of training module',
+    ),
+  );
+
+  return $permission;
+}
+
+function training_google_form($form, &$form_state){
+  $form = array();
+  $form['group'] = array( //группа элементов формы
+    '#type' => 'fieldset',
+    '#title' => 'group',
+    '#collapsible' => TRUE, //сворачиваемость
+    '#collapsed' => TRUE, //свернута по умолчанию
+
+  );
+  $form['group']['text'] = array(
+    '#type' => 'textfield',
+    '#title' => 'title',
+  );
+  $form['group']['text2'] = array(
+    '#type' => 'textfield',
+    '#title' => 'title2',
+    '#required' => TRUE,
+  );
+  $form['#attached']['css'] = array(
+    //путь к файлу
+  );
+  $form['submit'] = array(
+    '#type' => 'submit',
+    '#value' => 'ok'
+  );
+  return $form;
+}
+
+
+/**
+ * Implements hook_form_FORM_ID_alter
+ * @param $form
+ * @param $form_state
+ * @param $form_id
+ */
+function training_form_user_login_block_alter(&$form, &$form_state, $form_id ){
+
+  array_unshift($form['#validate'], 'secret_key_validate');
+  $form['#theme'] = 'test_form_theme_function';
+  $form['name']['#weight'] = -2;
+  $form['pass']['#weight'] = -1;
+  $form['secret_key'] = array(
+    '#type' => 'textfield',
+    '#title' => 'Secret Key',
+    '#size' => 15,
+    '#weight' => 0,
+  );
+  if(isset($form_state['input']['secret_key']) && $form_state['input']['secret_key'] == 'сим-сим откройся'){
+    $form['name']['#required'] = false;
+    $form['pass']['#required'] = false;
+  }
+  $form['#submit'][] = 'secret_key_submit';
+  $path = drupal_get_path('module','training') . '/css/training.css';
+  drupal_add_css($path);
```

Comment:
This way of adding CSS/JS works but this Drupal 6 way.
Better to use $form['#attached'] 
see https://api.drupal.org/api/drupal/developer%21topics%21forms_api_reference.html/7#attached